File: src-f/test_1.f90
program test_1
    use lin
    implicit none
    integer, parameter :: N=1000
    double precision, allocatable :: a(:,:), b(:)
    double precision :: maxerr
    integer :: i
    
    allocate(a(1:N,1:N))
    allocate(b(1:N))
    
    a(1:N,1:N)=1.0
    b(1:N)=1000.0
    do i=1, N
        a(i,i)=1001.0
    enddo
    call solve(a,b)
    maxerr=0.0
    do i=1, N
        if(abs(b(i)-0.5)>abs(maxerr))then
            maxerr=b(i)-0.5
        endif
    enddo
    write(*,'(e13.6)') maxerr
    if(abs(maxerr)>1.0e-8)then
        stop 1
    endif
    deallocate(a)
    deallocate(b)
end

File: src-java/test_1.java
import java.io.*;
import java.lang.Math.*;

class Test_1{
    public static void main(String arg[])
    {
	int n=1000;
	double[][] a=new double[n][n];
	double[] b=new double[n];
	Lin l=new Lin();

	for(int i=0; i<n; i++){
	    for(int j=0; j<n; j++){
		a[i][j]=1.0;
	    }
	    a[i][i]=1001.0;
	    b[i]=1000.0;
	}
	l.solve(a,b);

	double maxerr=0.0;
	for(int i=0; i<n; i++){
	    if(Math.abs(b[i]-0.5)>Math.abs(maxerr)){
		maxerr=b[i]-0.5;
	    }
	}
	System.out.println(maxerr);
    }
}
